spec/models/polling/cert_status_updater_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'rails_helper'
+require 'polling/cert_status_updater'
+
+RSpec.describe CertStatusUpdater, type: :model do
+
+  include CertStatusUpdater
+
+  let(:root) { PKI.new }
+
+  let(:msa_enc_cert_1) { create(:msa_encryption_certificate, in_use_at: nil) }
+  let(:msa_enc_cert_2) { create(:msa_encryption_certificate, in_use_at: nil) }
+  let(:msa_sgn_cert_1) { create(:msa_signing_certificate, in_use_at: nil) }
+  let(:msa_sgn_cert_2) { create(:msa_signing_certificate, in_use_at: nil) }
+  let(:msa_sgn_cert_3) { create(:msa_signing_certificate, in_use_at: nil) }
+
+  let(:service_1) { create(:service, entity_id: 'entity 1') }
+  let(:service_2) { create(:service, entity_id: 'entity 2') }
+  let(:service_3) { create(:service, entity_id: 'entity 3') }
+  let(:sp_component_1) { create(:sp_component, services: [service_1]) }
+  let(:sp_component_123) { create(:sp_component, services: [service_1, service_2, service_3]) }
+  let(:sp_enc_cert_for_single_entity) { create(:sp_encryption_certificate, component: sp_component_1, in_use_at: nil) }
+  let(:sp_enc_cert_for_multiple_entities) { create(:sp_encryption_certificate, component: sp_component_123, in_use_at: nil) }
+  let(:sp_sgn_cert_for_single_entity) { create(:sp_signing_certificate, component: sp_component_1, in_use_at: nil) }
+  let(:sp_sgn_cert_for_multiple_entities) { create(:sp_signing_certificate, component: sp_component_123, in_use_at: nil) }
+  let(:sp_sgn_cert_other) { create(:sp_signing_certificate, in_use_at: nil) }
+
+  describe '#update_hub_usage_status_for_cert' do
+    context 'When given an encryption cert belonging to an MSA component' do
+      it "leaves the cert unchanged if a call to the Hub does not show it is in use" do
+        allow_any_instance_of(HubConfigApi).to receive(:encryption_certificate).and_return(msa_enc_cert_2.x509)
+
+        expect(msa_enc_cert_1.in_use_at).to be(nil)
+        update_hub_usage_status_for_cert(msa_enc_cert_1)
+        expect(msa_enc_cert_1.in_use_at).to be(nil)
+      end
+
+      it "updates the cert's timestamp if a call to the Hub shows it is in use" do
+        allow_any_instance_of(HubConfigApi).to receive(:encryption_certificate).and_return(msa_enc_cert_1.x509)
+
+        expect(msa_enc_cert_1.in_use_at).to be(nil)
+        update_hub_usage_status_for_cert(msa_enc_cert_1)
+        expect(msa_enc_cert_1.in_use_at).not_to be(nil)
+      end
+    end
+
+    context 'When given a signing cert belonging to an MSA component' do
+      it "leaves the cert unchanged if a call to the Hub does not show it is in use" do
+        allow_any_instance_of(HubConfigApi).to receive(:signing_certificates).and_return([msa_sgn_cert_2.x509, msa_sgn_cert_3])
```

Comment:
Comment above about the return value of the HubConfigApi, in addition to that, I think [msa_sgn_cert_2.x509, msa_sgn_cert_3] is inconsistent. One of the certificate is x509 and the other is the actual Certificate Model here. I would expect they should be of the same kind

Reply:
You're right, I missed them.  Now fixed.